diffsync/helpers.py

```diff
@@ -339,42 +348,51 @@ def sync_diff_element(self, element: DiffElement, parent_model: "DiffSyncModel"
         # We only actually need the "new" attrs to perform a create/update operation, and don't need any for a delete
         attrs = diffs.get("+", {})
 
-        model: Optional["DiffSyncModel"]
+        # Retrieve Source Object to get its flags
+        src_model: Optional["DiffSyncModel"]
         try:
-            model = self.dst_diffsync.get(self.model_class, ids)
-            model.set_status(DiffSyncStatus.UNKNOWN)
+            src_model = self.src_diffsync.get(self.model_class, ids)
         except ObjectNotFound:
-            model = None
+            src_model = None
 
-        changed, modified_model = self.sync_model(model, ids, attrs)
-        model = modified_model or model
+        # Retrieve Dest (and primary) Object
+        dst_model: Optional["DiffSyncModel"]
+        try:
+            dst_model = self.dst_diffsync.get(self.model_class, ids)
+            dst_model.set_status(DiffSyncStatus.UNKNOWN)
+        except ObjectNotFound:
+            dst_model = None
+
+        changed, modified_model = self.sync_model(src_model=src_model, dst_model=dst_model, ids=ids, attrs=attrs)
+        dst_model = modified_model or dst_model
 
-        if not modified_model or not model:
+        if not modified_model or not dst_model:
             self.logger.warning("No object resulted from sync, will not process child objects.")
             return changed
 
-        if self.action == "create":
+        if self.action == "create":  # type: ignore
             if parent_model:
-                parent_model.add_child(model)
-            self.dst_diffsync.add(model)
+                parent_model.add_child(dst_model)
+            self.dst_diffsync.add(dst_model)
         elif self.action == "delete":
             if parent_model:
-                parent_model.remove_child(model)
-            if model.model_flags & DiffSyncModelFlags.SKIP_CHILDREN_ON_DELETE:
-                # We don't need to process the child objects, but we do need to discard them from the dst_diffsync
-                self.dst_diffsync.remove(model, remove_children=True)
+                parent_model.remove_child(dst_model)
+
+            skip_children = bool(dst_model.model_flags & DiffSyncModelFlags.SKIP_CHILDREN_ON_DELETE)
+            self.dst_diffsync.remove(dst_model, remove_children=skip_children)
+
+            if skip_children:
                 return changed
-            self.dst_diffsync.remove(model)
 
         self.incr_elements_processed()
 
         for child in element.get_children():
-            changed |= self.sync_diff_element(child, parent_model=model)
+            changed |= self.sync_diff_element(child, parent_model=dst_model)
 
         return changed
 
-    def sync_model(
-        self, model: Optional["DiffSyncModel"], ids: Mapping, attrs: Mapping
+    def sync_model(  # pylint: disable=too-many-branches
+        self, src_model: Optional["DiffSyncModel"], dst_model: Optional["DiffSyncModel"], ids: Mapping, attrs: Mapping
```

Comment:
Looks like `src_model` is an unused variable now, so can we revert the changes to this function?

Reply:
My preference would be to keep it in case we want to make a decision on a source flag in the future.
I think it was a miss in the design of this function

Reply:
Fair point.